L04-generics/src/main/java/homework/CustomerService.java

```diff
@@ -0,0 +1,37 @@
+package homework;
+
+import java.util.AbstractMap;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class CustomerService {
+
+    private TreeMap<Customer, String> treeMap = new TreeMap<>(Comparator.comparing(Customer::getScores));
```

Comment:
```suggestion
    private NavigableMap<Customer, String> treeMap = new TreeMap<>(Comparator.comparing(Customer::getScores));
```

Тип лучше объявлять минимальным интерфейсом.